local_db/data_access.py

```diff
@@ -188,3 +190,49 @@ def reject_file(self, request_id: str, relpath: UrlPath, username: str):
             )
             review.status = FileReviewStatus.REJECTED
             review.save()
+
+    def _create_audit_log(self, audit: AuditEvent):
+        event = AuditLog.objects.create(
+            type=audit.type,
+            user=audit.user,
+            workspace=audit.workspace,
+            request=audit.request,
+            path=audit.path,
+            extra=audit.extra,
+        )
+        return event
+
+    def audit_event(self, audit: AuditEvent):
+        with transaction.atomic():
+            self._create_audit_log(audit)
+
+    def get_audit_log(
+        self,
+        user: str | None = None,
+        workspace: str | None = None,
+        request: str | None = None,
+    ) -> list[AuditEvent]:
+        qs = AuditLog.objects.all().order_by("-created_at")
+
+        # TODO: we probably will need pagination?
+
+        if user:
+            qs = qs.filter(user=user)
+        if workspace:
+            qs = qs.filter(workspace=workspace)
+        if request:
+            qs = qs.filter(request=request)
```

Comment:
Does it ever make sense to ask for logs for both a workspace and a request?

Reply:
I guess not, as a request implied a workspace